test/datatypes/datetime.coffee

```diff
@@ -1515,12 +1528,12 @@ describe 'DateTime.sameOrBefore', ->
     DateTime.parse('2000-01').sameOrBefore(DateTime.parse('2000-02-01T00:00:00.0')).should.be.true()
     DateTime.parse('2000').sameOrBefore(DateTime.parse('2001-01-01T00:00:00.0')).should.be.true()
 
-  it 'should accept cases where a has unknown values but is still deterministicly before or same as b', ->
-    DateTime.parse('2000-01-01T00:00:00').sameOrBefore(DateTime.parse('2000-01-01T00:00:00.999')).should.be.true()
-    DateTime.parse('2000-01-01T00:00').sameOrBefore(DateTime.parse('2000-01-01T00:00:59.999')).should.be.true()
-    DateTime.parse('2000-01-01T00').sameOrBefore(DateTime.parse('2000-01-01T00:59:59.999')).should.be.true()
-    DateTime.parse('2000-01-01').sameOrBefore(DateTime.parse('2000-01-01T23:59:59.999')).should.be.true()
-    DateTime.parse('2000-01').sameOrBefore(DateTime.parse('2000-01-31T23:59:59.999')).should.be.true()
+  it 'should handle cases where a has unknown values but is not always deterministicly before or same as b', ->
```

Comment:
Actually, in all the cases below `a` _is_ deterministically before or same as `b` because `b` is always the _highest possible value_ in `a`'s uncertainty range.  There is no possible moment represented by `a` that comes _after_ `b`.

So... the old tests represented the _technically correct_ answers -- but the new tests represent the correct calculations _using the approach prescribed by the specification_.  Dang.

Reply:
Anyway, the tests are "right", but the description in the test case isn't exactly true, so consider changing it.